integrationtests/volume_test.go

```diff
@@ -177,3 +157,314 @@ func TestVolumeAPIs(t *testing.T) {
+		SizeBytes: newVolumeSize,
+	}
+
+	t.Logf("Attempt to resize volume from sizeBytes=%d to sizeBytes=%d", oldVolumeSize, newVolumeSize)
+
+	_, err = volumeClient.ResizeVolume(context.TODO(), resizeVolumeRequest)
+	if err != nil {
+		t.Fatalf("Volume resize request failed. Error: %v", err)
+	}
+
+	volumeStatsResponse, err = volumeClient.GetVolumeStats(context.TODO(), volumeStatsRequest)
+	if err != nil {
+		t.Fatalf("VolumeStats request after resize error: %v", err)
+	}
+	// resizing from 1GB to approximately 1.5GB
+	if !sizeIsAround(t, volumeStatsResponse.TotalBytes, newVolumeSize) {
+		t.Fatalf("VolumeSize reported should be greater than the old size, it is %v", volumeStatsResponse.TotalBytes)
+	}
+
+	volumeDiskNumberRequest := &volume.GetDiskNumberFromVolumeIDRequest{
+		VolumeId: volumeID,
+	}
+
+	volumeDiskNumberResponse, err := volumeClient.GetDiskNumberFromVolumeID(context.TODO(), volumeDiskNumberRequest)
+	if err != nil {
+		t.Fatalf("GetDiskNumberFromVolumeID failed: %v", err)
+	}
+
+	diskStatsRequest := &disk.GetDiskStatsRequest{
+		DiskNumber: volumeDiskNumberResponse.DiskNumber,
+	}
+
+	diskStatsResponse, err := diskClient.GetDiskStats(context.TODO(), diskStatsRequest)
+	if err != nil {
+		t.Fatalf("DiskStats request error: %v", err)
+	}
+
+	if diskStatsResponse.TotalBytes < 0 {
+		t.Fatalf("Invalid disk size was returned %v", diskStatsResponse.TotalBytes)
+	}
+
+	// Mount the volume
+	mountVolumeRequest := &volume.MountVolumeRequest{
+		VolumeId:   volumeID,
+		TargetPath: vhd.Mount,
+	}
+	_, err = volumeClient.MountVolume(context.TODO(), mountVolumeRequest)
+	if err != nil {
+		t.Fatalf("Volume id %s mount to path %s failed. Error: %v", volumeID, vhd.Mount, err)
+	}
+
+	// Unmount the volume
+	unmountVolumeRequest := &volume.UnmountVolumeRequest{
+		VolumeId:   volumeID,
+		TargetPath: vhd.Mount,
+	}
+	_, err = volumeClient.UnmountVolume(context.TODO(), unmountVolumeRequest)
+	if err != nil {
+		t.Fatalf("Volume id %s mount to path %s failed. Error: %v", volumeID, vhd.Mount, err)
+	}
+}
+
+func volumeTests(t *testing.T) {
+	volumeClient, err := volume.New(volumeapi.New())
+	require.Nil(t, err)
+
+	diskClient, err := disk.New(diskapi.New())
+	require.Nil(t, err)
+
+	t.Run("MountVolume", func(t *testing.T) {
+		mountVolumeTests(diskClient, volumeClient, t)
+	})
+	t.Run("GetClosestVolumeFromTargetPath", func(t *testing.T) {
+		getClosestVolumeFromTargetPathTests(diskClient, volumeClient, t)
+	})
+}
+
+func TestVolume(t *testing.T) {
+	t.Run("NegativeDiskTests", func(t *testing.T) {
+		negativeDiskTests(t)
+	})
+	t.Run("NegativeVolumeTests", func(t *testing.T) {
+		negativeVolumeTests(t)
+	})
+
+	// TODO: These tests will fail on Github Actions because Hyper-V is disabled
+	// see https://github.com/actions/virtual-environments/pull/2525
+
+	// these tests should be considered frozen from the API point of view
+	t.Run("volumeTests", func(t *testing.T) {
+		skipTestOnCondition(t, isRunningOnGhActions())
+		volumeTests(t)
```

Comment:
I think we could include the contents of volumeTests directly here, that way we don't have `TestVolume` and `volumeTests`